src/consumerIncentives/selectors.ts

```diff
@@ -0,0 +1,60 @@
+import { createSelector } from 'reselect'
+import { superchargeTokenConfigByTokenSelector } from 'src/app/selectors'
+import { SuperchargeTokenConfig } from 'src/consumerIncentives/types'
+import { tokensByAddressSelector } from 'src/tokens/selectors'
+import Logger from 'src/utils/Logger'
+
+interface SuperchargeInfo {
+  hasBalanceForSupercharge: boolean
+  superchargingTokenConfig?: SuperchargeTokenConfig
+  hasMaxBalance?: boolean
+  superchargeBalance?: number
+  superchargeUsdBalance?: number
+}
+
+// Supercharge info for the token with the highest USD balance among the possible supercharge tokens for which the minimum token balance is met
+// Note: this logic should match what's done on the backend to determine the supercharging tokens
+export const superchargeInfoSelector = createSelector(
+  tokensByAddressSelector,
+  superchargeTokenConfigByTokenSelector,
+  (tokensByAddress, superchargeTokenConfigByToken): SuperchargeInfo => {
+    let superchargeInfo: SuperchargeInfo = { hasBalanceForSupercharge: false }
+
+    for (const [tokenAddress, superchargeTokenConfig] of Object.entries(
+      superchargeTokenConfigByToken
+    )) {
+      const tokenInfo = tokensByAddress[tokenAddress.toLowerCase()]
+      if (!tokenInfo || !tokenInfo.usdPrice) {
+        Logger.warn(`A possible supercharged token info couldn't be obtained from firebase`)
```

Comment:
not a big issue but i think this might get logged a lot with the super analytics prop - the chances of tokenInfo being falsey are low but the `usdPrice` on the tokenInfo is set to null when they become stale (12 hours) so potentially this could get logged on every app open if the user visits the app once a day. maybe we can move `!tokenInfo.usdPrice` to the next if statement that doesn't contain the logging?

Reply:
Yes, I thought this could be logged several times, I think I'm removing this log entirely.